infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/WaitSnapshotApplyState.java

```diff
@@ -73,14 +73,16 @@ public WaitSnapshotApplyState(LogReplicationFSM logReplicationFSM, DataSender da
     public LogReplicationState processEvent(LogReplicationEvent event) throws IllegalTransitionException {
         switch (event.getType()) {
             case SNAPSHOT_SYNC_REQUEST:
-                log.info("Snapshot Sync requested {} while waiting for {} to complete.", event.getEventID(), getTransitionEventId());
+                log.info("Snapshot Sync requested {} while waiting for {} to complete.", event.getEventId(), getTransitionEventId());
                 LogReplicationState snapshotSyncState = fsm.getStates().get(LogReplicationStateType.IN_SNAPSHOT_SYNC);
-                snapshotSyncState.setTransitionEventId(event.getEventID());
+                snapshotSyncState.setTransitionEventId(event.getEventId());
+                ((InSnapshotSyncState)snapshotSyncState).setForcedSnapshotSync(event.getMetadata().isForcedSnapshotSync());
                 return snapshotSyncState;
             case SYNC_CANCEL:
                 log.debug("Sync has been canceled while waiting for Snapshot Sync {} to complete apply. Restart.", transitionEventId);
                 LogReplicationState inSnapshotSyncState = fsm.getStates().get(LogReplicationStateType.IN_SNAPSHOT_SYNC);
                 inSnapshotSyncState.setTransitionEventId(UUID.randomUUID());
+                ((InSnapshotSyncState)inSnapshotSyncState).setForcedSnapshotSync(false);
```

Comment:
is forceSnapshotSync guaranteed to be false?

Reply:
In the case of SYNC_CANCEL, it will automatically start a new round of snapshot sync, which means it is not forced.